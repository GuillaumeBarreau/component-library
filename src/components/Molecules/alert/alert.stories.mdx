import { Meta, Story, Canvas, ArgsTable } from "@storybook/addon-docs";
import { Alert } from "./alert";

<Meta 
  title="Components/Molecules/Alert" 
  component={Alert}
  argTypes={{
    notice: {
      control: "select",
      description: "Modify the Alert background",
      options: [
        "default",
        "info",
        "warning",
        "success",
        "error",
      ],
    },
    progressBar: {
      description: "show process progressBar",
      control: "number",
    },
    progressBarLabel: {
      description: "show process progressBar",
      control: "number",
    },
    className: {
      description: "Override a component's styles",
      control: "text",
    },
    children: {
        control: "text",
        description: "Text content",
    }
  }}
  args={{
    children: 'children',
    handleClick: () => null
  }}/>

# Alert

The Alert component is a small text box notifying the user with short, timely, and relevant information. 
The Alert component appears at the bottom-right of the page.

## React Component

#### Import library

```
Import {Molecules} from '@bonjour-les-pixels/component-library';
```

#### Use Buttom component

```
<Molecules.Alert {...props}> CHILDREN_TEXT </Molecules.Alert>
```

## Playground

<Canvas>
  <Story name="Alert">
    {args => {
      return <Alert {...args}  />;
    }}
  </Story>
</Canvas>

### Props

<ArgsTable story="Alert" />

### Notice

You can use the `notice` prop on a Alert to modify the background color. You can set the value to primary, success, warning, info or error. The style varies according to the level of importance of the action in the system context.

<Canvas>
  <Story name="Alert/Notice">
    <Alert
      size="medium"
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="success"
      size="medium"
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="info"
      size="medium"
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="error"
      size="medium"
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="warning"
      size="medium"
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
  </Story>
</Canvas>

### ClassName

You can use the `className` prop to override a component's styles using custom classes.

```
  <Alert className="OVERRIDE_STYLE_CLASSNAME" />
```

### HandleClick

You can use the prop `HandleClick` to display the action button to trigger a window closure.

<Canvas>
  <Story name="Alert/HandleClick">
    <Alert
      size="medium"
      handleClick={() => null}
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="success"
      size="medium"
      handleClick={() => null}
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="info"
      size="medium"
      handleClick={() => null}
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="error"
      size="medium"
      handleClick={() => null}
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="warning"
      size="medium"
      handleClick={() => null}
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
  </Story>
</Canvas>

### ProgressBar

You can use the `progressBar` props to display the progress bar time until the window is automatically closed.

<Canvas>
  <Story name="Alert/ProgressBar">
    <Alert
      size="medium"
      progressBar={10}
      handleClick={() => null}
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="success"
      handleClick={() => null}
      size="medium"
      progressBar={75}
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="info"
      handleClick={() => null}
      size="medium"
      progressBar={25}
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="error"
      handleClick={() => null}
      size="medium"
      progressBar={50}
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="warning"
      handleClick={() => null}
      size="medium"
      progressBar={75}
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
  </Story>
</Canvas>

### ProgressBarLabel

You can use the boolean `progressBarLabel` props to display a label (example: 46%).

<Canvas>
  <Story name="Alert/ProgressBarLabel">
    <Alert
      size="medium"
      progressBar={10}
      handleClick={() => null}
      progressBarLabel
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="success"
      handleClick={() => null}
      progressBarLabel
      size="medium"
      progressBar={75}
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="info"
      handleClick={() => null}
      progressBarLabel
      size="medium"
      progressBar={25}
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="error"
      handleClick={() => null}
      progressBarLabel
      size="medium"
      progressBar={50}
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="warning"
      handleClick={() => null}
      progressBarLabel
      size="medium"
      progressBar={75}
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
  </Story>
</Canvas>
