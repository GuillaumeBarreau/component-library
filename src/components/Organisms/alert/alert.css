/* alert */
.alert {
  display: flex;
  flex-direction: column;
  border: 0.125rem solid;
  border-radius: 0.25rem;
  padding: 0.25rem;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.alert--content-top {
  padding: 0.5rem 1rem;
  display: flex;
  align-items: center;
}

.alert--content-top > svg {
  margin-right: 1rem;
}

.alert--content-right {
  height: 100%;
  display: flex;
  align-items: center;
  padding-left: 1rem;
  margin-left: 1rem;
  justify-content: center;
}

.alert--content-right > button > svg {
  width: 1.5rem;
  height: 1.5rem;
}

.alert--content-bottom {
  display: flex;
  align-items: center;
}

.alert--overide-typography {
  margin: 0rem 1rem 0rem 1.5rem;
  min-width: 35px;
  text-align: center;
}

.alert--notice-default {
  border-color: var(--color-text--light);
  background: #f0f0f0;
}

.alert--notice-default > .alert--content-top > .alert--content-right {
  border-left: 0.0625rem solid var(--color-text--light);
}

.alert--notice-warning > .alert--content-top > .alert--content-right {
  border-left: 0.0625rem solid var(--color-notice-warning--dark);
}

.alert--notice-success > .alert--content-top > .alert--content-right {
  border-left: 0.0625rem solid var(--color-notice-success--dark);
}

.alert--notice-error > .alert--content-top > .alert--content-right {
  border-left: 0.0625rem solid var(--color-notice-error--dark);
}

.alert--notice-info > .alert--content-top > .alert--content-right {
  border-left: 0.0625rem solid var(--color-notice-info--dark);
}

.alert--notice-success {
  border-color: var(--color-notice-success--dark);
  background: #c3e9d3;
}

.alert--notice-error {
  border-color: var(--color-notice-error--dark);
  background: #eeb5b5;
}

.alert--notice-info {
  border-color: var(--color-notice-info--dark);
  background: #cfe2ff;
}

.alert--notice-warning {
  border-color: var(--color-notice-warning--dark);
  background: #f0d1b6;
}

.alert--notice-default svg {
  color: var(--color-text--light);
}

.alert--notice-success svg {
  color: var(--color-notice-success--dark);
}

.alert--notice-error svg {
  color: var(--color-notice-error--dark);
}

.alert--notice-info svg {
  color: var(--color-notice-info--dark);
}

.alert--notice-warning svg {
  color: var(--color-notice-warning--dark);
}

.alert--status-fadeOut {
  display: none;
}
