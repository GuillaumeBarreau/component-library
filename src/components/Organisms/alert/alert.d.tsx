import { INotice } from "components/_settings/globalTypes/Types.d";

export type IProgressBarProps =
  | { progressBar?: number; progressBarLabel?: never }
  | { progressBar: number; progressBarLabel?: boolean }
  | { progressBarLabel: true; progressBar: number };

export interface ComponentAlertProps {
  children: React.ReactNode;
  className?: string;
  dismissible?: boolean;
  notice?: INotice;
}
