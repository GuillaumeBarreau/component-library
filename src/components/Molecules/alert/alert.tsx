import React, { useState } from "react";
import classNames from "classnames";
import { ComponentAlertProps, IProgressBarProps } from "./alert.d";
import { Icon } from "components/Atoms/icon";
import { Typography } from "components/Atoms/typography";
import { Button } from "components/Atoms/button";
import { ProgressBar } from "components/Atoms/progressBar";
import "./alert.css";
import { TiconName } from "components/_settings/globalTypes/Types.d";

export const Alert: React.FC<ComponentAlertProps & IProgressBarProps> = (
  props
) => {
  const {
    children,
    className,
    notice = "default",
    progressBar,
    progressBarLabel,
    handleClick,
  } = props;

  const prefixCls = "alert";

  const classes = classNames(
    prefixCls,
    {
      [`${prefixCls}--notice-${notice}`]: notice,
    },
    className
  );

  const iconAlert: {
    [key: string]: TiconName;
  } = {
    info: "bell",
    warning: "circle-exclamation",
    success: "circle-check",
    error: "bomb",
    default: "coffee",
  };

  return (
    <div className={classes}>
      <div className="alert--content-top">
        <Icon icon={iconAlert[notice]} size="sm" />
        <Typography className={"alert--overide-typography"} size="small">
          {children}
        </Typography>
        {handleClick && (
          <div className="alert--content-right">
            <Button handleClick={handleClick} iconName="circle-xmark" />
          </div>
        )}
      </div>
      {(progressBar || progressBar === 0) && (
        <div className="alert--content-bottom">
          <ProgressBar progress={progressBar} size="xsmall" notice={notice} />
          {progressBarLabel && (
            <Typography className="alert--overide-typography">
              {progressBar}%
            </Typography>
          )}
        </div>
      )}
    </div>
  );
};
