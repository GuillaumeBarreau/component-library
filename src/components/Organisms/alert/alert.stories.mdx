import { Meta, Story, Canvas, ArgsTable } from "@storybook/addon-docs";
import { Alert } from "./alert";

<Meta 
  title="Components/Organisms/Alert" 
  component={Alert}
  argTypes={{
    notice: {
      control: "select",
      description: "Modify the Alert background",
      options: [
        "default",
        "info",
        "warning",
        "success",
        "error",
      ],
    },
    progressBar: {
      description: "show process progressBar",
      control: "number",
    },
    progressBarLabel: {
      description: "show process progressBar",
      control: "number",
    },
    className: {
      description: "Override a component's styles",
      control: "text",
    },
    children: {
        control: "text",
        description: "Text content",
    }
  }}
  args={{
    children: 'children',
    dismissible: 'true'
  }}/>

# Alert

The Alert component is a small text box notifying the user with short, timely, and relevant information. 
The Alert component appears at the bottom-right of the page.

## React Component

#### Import library

```
Import {Organisms} from '@bonjour-les-pixels/component-library';
```

#### Use Buttom component

```
<Organisms.Alert {...props}> CHILDREN_TEXT </Organisms.Alert>
```

## Playground

<Canvas>
  <Story name="Alert">
    {args => {
      return <Alert {...args}  />;
    }}
  </Story>
</Canvas>

### Props

<ArgsTable story="Alert" />

### Notice

You can use the `notice` prop on a Alert to modify the background color. You can set the value to primary, success, warning, info or error. The style varies according to the level of importance of the action in the system context.

<Canvas>
    <Alert
      size="medium"
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="success"
      size="medium"
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="info"
      size="medium"
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="error"
      size="medium"
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="warning"
      size="medium"
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
</Canvas>

### ClassName

You can use the `className` prop to override a component's styles using custom classes.

```
  <Alert className="OVERRIDE_STYLE_CLASSNAME" />
```

### Dismissible

You can use the prop `dismissible` to display the action button to trigger a window closure.

<Canvas>
    <Alert
      size="medium"
      dismissible
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="success"
      dismissible
      size="medium"
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="info"
      dismissible
      size="medium"
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="error"
      dismissible
      size="medium"
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="warning"
      dismissible
      size="medium"
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
</Canvas>

### ProgressBar

You can use the `progressBar` props to display the progress bar time until the window is automatically closed.

<Canvas>
    <Alert
      size="medium"
      progressBar={10}
      dismissible
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="success"
      dismissible
      size="medium"
      progressBar={75}
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="info"
      dismissible
      size="medium"
      progressBar={25}
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="error"
      dismissible
      size="medium"
      progressBar={50}
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="warning"
      dismissible
      size="medium"
      progressBar={75}
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
</Canvas>

### ProgressBarLabel

You can use the boolean `progressBarLabel` props to display a label (example: 46%).

<Canvas>
    <Alert
      size="medium"
      progressBar={10}
      dismissible
      progressBarLabel
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="success"
      dismissible
      progressBarLabel
      size="medium"
      progressBar={75}
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="info"
      dismissible
      progressBarLabel
      size="medium"
      progressBar={25}
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="error"
      dismissible
      progressBarLabel
      size="medium"
      progressBar={50}
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
    <Alert
      notice="warning"
      dismissible
      progressBarLabel
      size="medium"
      progressBar={75}
      children="Lorem, ipsum dolor sit amet consectetur adipisicing elit. Labore ut in incidunt ea nostrum, voluptatem explicabo voluptas."
    />
</Canvas>
