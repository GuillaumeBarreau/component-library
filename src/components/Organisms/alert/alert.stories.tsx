import { Alert } from './alert';
import { Meta } from "@storybook/react/types-6-0";
import { ComponentAlertProps } from "./alert.d";

export default {
  title: 'Alert',
  component: Alert,
  parameters: {
    a11y: {
      // optional selector which element to inspect
      element: "#root",
      // optional flag to prevent the automatic check
      manual: false,
    },
  },
} as Meta;

const Template: any = (
  args: JSX.IntrinsicAttributes & ComponentAlertProps
) => <>
   <section style={ {display: "flex", justifyContent: "center"} }>
    <Alert {...args} /> 
  </section>
</>;

export const AlertDefault = Template.bind({});
AlertDefault.args = {
  children: "alert",
};
