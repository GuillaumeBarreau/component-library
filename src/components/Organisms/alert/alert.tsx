import React, { useState } from "react";
import classNames from "classnames";
import { ComponentAlertProps, IProgressBarProps } from "./alert.d";
import { Icon } from "components/Atoms/icon";
import { Typography } from "components/Atoms/typography";
import { Button } from "components/Atoms/button";
import { ProgressBar } from "components/Atoms/progressBar";
import "./alert.css";
import { IconProp } from "@fortawesome/fontawesome-svg-core";

export const Alert: React.FC<ComponentAlertProps & IProgressBarProps> = (
  props
) => {
  const [fadeOut, setFadeOut] = useState(false);
  const {
    children,
    className,
    notice = "default",
    dismissible,
    progressBar,
    progressBarLabel,
  } = props;

  const prefixCls = "alert";

  const classes = classNames(
    prefixCls,
    {
      [`${prefixCls}--notice-${notice}`]: notice,
      [`${prefixCls}--status-fadeOut`]: fadeOut || progressBar === 100,
    },
    className
  );

  const iconAlert: {
    [key: string]: IconProp;
  } = {
    info: "bell",
    warning: "circle-exclamation",
    success: "circle-check",
    error: "bomb",
    default: "coffee",
  };

  return (
    <div className={classes}>
      <div className="alert--content-top">
        <Icon icon={`${iconAlert[notice]}`} size="sm" />
        <Typography className={"alert--overide-typography"} size="small">
          {children}
        </Typography>
        {dismissible && (
          <div className="alert--content-right">
            <Button
              handleClick={() => {
                setFadeOut(true);
              }}
              iconName="circle-xmark"
            />
          </div>
        )}
      </div>
      {(progressBar || progressBar === 0) && (
        <div className="alert--content-bottom">
          <ProgressBar progress={progressBar} size="xsmall" notice={notice} />
          {progressBarLabel && (
            <Typography className="alert--overide-typography">
              {progressBar}%
            </Typography>
          )}
        </div>
      )}
    </div>
  );
};
