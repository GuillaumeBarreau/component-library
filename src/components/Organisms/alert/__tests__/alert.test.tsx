import React from "react";
import { shallow } from "enzyme";
import { Alert } from "../alert";
import { ComponentAlertProps } from "../alert.d";

describe("Alert", () => {
  let defaultProps: JSX.IntrinsicAttributes & ComponentAlertProps;

  beforeEach(() => {
    defaultProps = {
      children: "children Content",
      className: "override-class",
      dismissible: true,
      progressBar: 64,
      notice: "info",
    };
  });

  it("renders correctly", () => {
    const wrapper = shallow(<Alert {...defaultProps} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("mount correctly", () => {
    const wrapper = shallow(<Alert {...defaultProps} />);
    expect(() => wrapper).not.toThrow();
  });

  it("Should contain defaultProps:children correctly", () => {
    const wrapper = shallow(<Alert {...defaultProps} />);
    expect(() => wrapper).not.toThrow();
    expect(wrapper.html()).toContain("children Content");
  });

  it("Should contain defaultProps:className correctly", () => {
    const wrapper = shallow(<Alert {...defaultProps} />);
    expect(wrapper.html()).toContain("override-class");
  });

  it("Should contain defaultProps:progressBar correctly", () => {
    const wrapper = shallow(<Alert {...defaultProps} />);
    expect(wrapper.html()).toContain("width:64%");
  });

  it("Should contain defaultProps:notice correctly", () => {
    const wrapper = shallow(<Alert {...defaultProps} />);
    expect(wrapper.find(".alert--notice-info")).toHaveLength(1);
  });

  it("Should contain props:notice correctly with new value", () => {
    const wrapper = shallow(<Alert {...defaultProps} notice="warning" />);
    expect(wrapper.find(".alert--notice-warning")).toHaveLength(1);
  });

  it("Should contain defaultProps:dismissible a new tag html", () => {
    const wrapper = shallow(<Alert {...defaultProps} />);
    expect(wrapper.find(".alert--content-right")).toHaveLength(1);
  });

  it("Should not appear a new tag html with Props:dismissible", () => {
    const wrapper = shallow(<Alert {...defaultProps} dismissible={false} />);
    expect(wrapper.find(".alert--content-right")).toHaveLength(0);
  });
});
