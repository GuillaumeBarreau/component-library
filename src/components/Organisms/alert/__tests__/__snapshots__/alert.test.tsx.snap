// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders correctly 1`] = `
"<div className=\\"alert alert--notice-info override-class\\">
  <div className=\\"alert--content-top\\">
    <Icon icon=\\"bell\\" size=\\"sm\\" />
    <Typography className=\\"alert--overide-typography\\" size=\\"small\\">
      children Content
    </Typography>
    <div className=\\"alert--content-right\\">
      <Button handleClick={[Function: handleClick]} iconName=\\"circle-xmark\\" />
    </div>
  </div>
  <div className=\\"alert--content-bottom\\">
    <ProgressBar progress={64} size=\\"xsmall\\" notice=\\"info\\" />
  </div>
</div>"
`;
